libmamba/include/mamba/specs/repo_data.hpp

```diff
@@ -0,0 +1,238 @@
+        std::vector<std::string> depends = {};
+
+        /**
+         * Additional constraints on packages.
+         *
+         * `constrains` are different from `depends` in that packages specified in `depends` must
+         * be installed next to this package, whereas packages specified in `constrains` are not
+         * required to be installed, but if they are installed they must follow these constraints.
+         */
+        std::vector<std::string> constrains = {};
+
+        /**
+         * Track features are nowadays only used to deprioritize packages.
+         *
+         * To that effect, the number of track features is counted (number of commas) and the
+         * package is downweighted by the number of track_features.
+         */
+        std::vector<std::string> track_features = {};
+
+        /**
+         * Features are a deprecated way to specify different feature sets for the conda solver.
+         *
+         * This is not supported anymore and should not be used.
+         * Instead, `mutex` packages should be used to specify mutually exclusive features.
+         */
+        std::optional<std::string> features = {};
+
+        /** If this package is independent of architecture this field specifies in what way. */
+        std::optional<NoArchType> noarch = {};
+
+        /** The specific license of the package. */
+        std::optional<std::string> license = {};
+
+        /** The license family. */
+        std::optional<std::string> license_family = {};
+
+        /**
+         * The UNIX Epoch timestamp when this package was created.
+         *
+         * Note that sometimes this is specified in seconds and sometimes in milliseconds.
+         */
+        std::optional<std::size_t> timestamp = {};
+    };
+
+    /**
+     * Serialize to JSON.
+     *
+     * Optional members are omitted from json.
+     */
+    void to_json(nlohmann::json& j, const PackageRecord& p);
+    void to_json(nlohmann::json& j, PackageRecord&& p);
+
+    /**
+     * Deserialize from JSON
+     *
+     * Missing json entries fill optionals with a null values and collections as empty.
+     * Special handling of the following fields is performed:
+     * - ``"noarch"`` can be a string or a boolean (old behaviour), in which case ``false``
+     *   parse to ``std::nullopt`` and ``true`` to ``NoArchType::Generic``.
+     * - ``"track_features"`` can be a string or list of string. In the former case, it is
+     *   considered as a single element list.
+     */
+    void from_json(const nlohmann::json& j, PackageRecord& p);
+    void from_json(nlohmann::json&& j, PackageRecord& p);
+
+
+    /** Information about subdirectory of channel in the Conda RepoData. */
+    struct ChannelInfo
+    {
+        /** The channel's subdirectory. */
+        std::string subdir = {};
+    };
+
+    /** Serialize to JSON. */
+    void to_json(nlohmann::json& j, const ChannelInfo& info);
+    void to_json(nlohmann::json& j, ChannelInfo&& info);
```

Comment:
A const reference can bind to an rvalue, so not sure that we need this overload.